src/index.ts

```diff
@@ -126,29 +134,30 @@ export function targetConstructorToSchema(
  * Return `storage.validationMetadatas` populated with `constraintMetadatas`.
  */
 function getMetadatasFromStorage(
-  storage: cv.MetadataStorage
-): ValidationMetadata[] {
-  const metadatas: ValidationMetadata[] = (storage as any).validationMetadatas
-  return populateMetadatasWithConstraints(storage, metadatas)
+  storage: IStorage
+): Map<any, ValidationMetadata[]> {
```

Comment:
Can we keep the return type as a flat array `ValidationMetadata[]`? That would avoid the extra flattening step down the line in `validationMetadatasToSchemas`.